### PR TITLE
DOC Add links to GitHub and issue raising for every example

### PR DESCRIPTION
**Pull request by TamaraAtanasoska**
## Description
Fixes https://github.com/fairlearn/fairlearn/issues/1301 and closes https://github.com/fairlearn/fairlearn/pull/1329.
I spent some time looking into how I can automate this and make it generate links for each example, but this was a much easier solution for now.

@romanlutz and @hildeweerts please take a look!

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [x] example notebook added or updated

## Screenshots

Screenshot 2024-10-29 at 14 50 16

Screenshot 2024-10-29 at 14 50 37

### CODE REVIEW

**Comment by hildeweerts** [comment]
It's not like we're adding new example notebooks very often, so I think this solution is fine for now.

It would be nice if the created issue follows one of the issue templates, probably the documentation improvement one makes the most sense. I think that should be pretty easy to do by adding something like `labels=Documentation&projects=&template=doc_improvement.md` to each link.

I'm not a huge fan of the double note on top (see screengrab below), but I don't think there's an easy way to resolve that as the first one is generated automatically.

image

**Comment by TamaraAtanasoska** [comment]
@hildeweerts I can edit it to fit the template, that is a great remark.

**Review by adrinjalali — COMMENTED** [review]
Interesting one.

I guess, why not. I like how in sklearn you just have the buttons at the bottom of each example.

So, I'm kinda, could do, could not do, on this PR. I'll let others decide.

**Comment by TamaraAtanasoska** [comment]
> Interesting one.
> 
> I guess, why not. I like how in sklearn you just have the buttons at the bottom of each example.
> 
> So, I'm kinda, could do, could not do, on this PR. I'll let others decide.

I also like that the right sidebar carries a lot of the buttons and relevant info in scikit-learn

**Comment by TamaraAtanasoska** [comment]
@hildeweerts what do you think about moving this information to the sidebar on the right where the sections are? This way it  will be available as you scroll down all the time.

**Comment by hildeweerts** [comment]
TIL scikit-learn has moved to the pydata theme! @TamaraAtanasoska I like having all the buttons in the sidebar on the right, let's do it :)

**Comment by TamaraAtanasoska** [comment]
I found a better way than doing this manually, however I can't seem to make my setup work with it. I created an issue here: https://github.com/pydata/pydata-sphinx-theme/issues/2027.

**Comment by TamaraAtanasoska** [comment]
@romanlutz I seem to have found a way to do this :) 

Changes:
- there is a link on the right side not that leads to a new issue with the doc template as @hildeweerts proposed
- all the links are on the sidebar now, and there is no need for any notes (consistent with scikit-learn cc @adrinjalali)
- the bottom downloads are hidden as they aren't necessary any more

I really like how it looks now, it is more "clean". 

Screenshot 2024-11-04 at 09 22 03

**Comment by adrinjalali** [comment]
happy to review when things go green again 😁

**Comment by TamaraAtanasoska** [comment]
> happy to review when things go green again 😁

all green, versioning issue :)

**Comment by TamaraAtanasoska** [comment]
@adrinjalali (I replied to the comment a few hours ago, but maybe that doesn't ping :S)

File: docs/templates/edit-example-link.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="edit-example-link">
+    {% if pagename.split('/')[-1].startswith('plot_') %}
+        <a href="https://github.com/fairlearn/fairlearn/edit/main/examples/{{ pagename.split('/')[1]}}.py">
+            Edit this example on GitHub
+        </a>
+        <a href="https://github.com/fairlearn/fairlearn/issues/new?assignees=&labels=Documentation&template=doc_improvement.md&title=DOC%20Issue%20in%20{{ pagename.split('/')[1] }}&body=https://github.com/fairlearn/fairlearn/edit/main/examples/{{ pagename.split('/')[1] }}.py">
```
Comment (hildeweerts) [inline docs/templates/edit-example-link.html:6]: Currently, the template isn't used because the body is specified explicitly. I personally prefer having the template over the exact link because it encourages people to write something meaningful (and having the title point to the example name seems sufficient). We also need to use lowercase `documentation`, otherwise it doesn't work. So i'd suggest something like this:
 
```suggestion
        <a href="https://github.com/fairlearn/fairlearn/issues/new?labels=documentation&template=doc_improvement.md&title=DOC%20Issue%20in%20Example%20Notebook%20{{ pagename.split('/')[1] }}.py>
```

Which should result in a link like this: https://github.com/fairlearn/fairlearn/issues/new?labels=documentation&template=doc_improvement.md&title=DOC%20Issue%20in%20Example%20Notebook%20plot_quickstart_counts.py

**Review by adrinjalali — APPROVED** [review]
